p2p_beef_tx.go

```diff
@@ -1,16 +1,22 @@
 package paymail
 
 import (
+	"context"
 	"encoding/hex"
 	"errors"
 	"fmt"
-
+	"github.com/libsv/bitcoin-hc/transports/http/endpoints/api/merkleroots"
+	"github.com/libsv/go-bc"
 	"github.com/libsv/go-bt/v2"
+	"github.com/libsv/go-bt/v2/bscript/interpreter"
 )
 
-type CompoundMerklePath []map[string]uint64
-
-type CMPSlice []CompoundMerklePath
+type MerkleRootVerifier interface {
```

Comment:
I know I'm a pain in the ass. But... :D
Are you sure SPV is a part and responsibility of our decoded BEEF type?
We do SPV on the  BEEF data, but it's not up to it how we do it. I see an analogy here to the `string` type and pattern searching. It is known that we look for patterns in a `string`, but we use the regex package for this not string's methods.

To sum up: IMO all code about SPV should be placed in a file like `p2p_spv.go`. `MerkleRootVerifier` interface should be part of it, `ExecuteSimplifiedPaymentVerification` should be part of it and take `DecodedBEEF` as an argument, all validation methods should be written in the same way. `func (dBeef *DecodedBEEF) GetMerkleRoots()` should stay where it is ;-)

Reply:
Done